components/calendar/src/any_calendar.rs

```diff
@@ -739,6 +880,10 @@ impl AnyCalendarKind {
             b"chinese" => AnyCalendarKind::Chinese,
             b"coptic" => AnyCalendarKind::Coptic,
             b"iso" => AnyCalendarKind::Iso,
+            b"islamic" => AnyCalendarKind::IslamicObservational,
+            b"islamic_civil" => AnyCalendarKind::IslamicCivil,
+            b"islamic_umalqura" => AnyCalendarKind::UmmAlQura,
+            b"islamic_tbla" => AnyCalendarKind::IslamicTabular,
```

Comment:
Issue: Please be self-consistent with the BCP 47 byte strings. Use islamic, islamocc, umalqura, and tbla everywhere.

Reply:
I realize this is the exact opposite of @Manishearth's suggestion

Reply:
yes, we can discuss this part tomorrow together